### PR TITLE
Refactor Huggingface module url

### LINKED ISSUE #2576: [feature] Optionally provide Huggingface endpoint URL/path in `text2vec-huggingface`
The [Huggingface endpoint in t2v-hf](https://github.com/weaviate/weaviate/blob/abf30eac8656cf3bf1b471d9eb1416695777a53d/modules/text2vec-huggingface/clients/url.go) is hardcoded to the public HF endpoint.

Azure (and possibly others) offer [hosted endpoints](https://huggingface.co/blog/hugging-face-endpoints-on-azure), so making this URL and path optionally user-configurable would allow arbitrary deployment endpoints to be used.

This would provide appreciable cost and security benefits for users.

@hsm207

### PR DESCRIPTION
**Pull request by gmanninglive**
Closes: #2576
> The [Huggingface endpoint in t2v-hf](https://github.com/weaviate/weaviate/blob/abf30eac8656cf3bf1b471d9eb1416695777a53d/modules/text2vec-huggingface/clients/url.go) is hardcoded to the public HF endpoint.
> 
> Azure (and possibly others) offer [hosted endpoints](https://huggingface.co/blog/hugging-face-endpoints-on-azure), so making this URL and path optionally user-configurable would allow arbitrary deployment endpoints to be used.

### What's being changed:

I started off adding extra config variables for origin and path, see first commit. Then I realized there is already a config variable that has this functionality, at least as far as I can tell `EndpointURL`

Instead I have refactored the module to keep the same functionality but hopefully make it clearer.

I felt the builder pattern was superfluous as it only has two branches based on the config, Custom URL or Huggingface default. Instead I have refactored to a function `getHuggingFaceUrl` that keeps all url setting logic in one place.

Again this PR doesn't add any new functionality but hopefully closes #2576.

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [x] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

**Comment by gmanninglive** [comment]
@weaviate-git-bot I agree to CLA

**Comment by gmanninglive** [comment]
@kcm let me know if you think this closes the issue

File: modules/text2vec-huggingface/clients/vectorizer.go [hunk 1]
```diff
@@ -48,7 +48,7 @@ type huggingFaceApiError struct {
 type vectorizer struct {
 	apiKey                string
 	httpClient            *http.Client
-	urlBuilder            *huggingFaceUrlBuilder
+	url                   string
```
[the remark this answers is not in the document]
Comment (gmanninglive) [reply modules/text2vec-huggingface/clients/vectorizer.go]: I've just pushed a change that's not quite what you requested but I think it cleans things up even more moving `getHuggingFaceUrl` logic into the `getURL` method

**Review by antas-marcin — APPROVED** [review]

File: modules/text2vec-huggingface/clients/vectorizer.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (antas-marcin) [reply modules/text2vec-huggingface/clients/vectorizer.go]: looks good I will merge your PR. Thanks!

**Pull request merged**